pkg/transaction/sender_matcher.go

```diff
@@ -71,35 +71,62 @@ func (m *Matcher) Matches(ctx context.Context, tx []byte, networkID uint64, send
 		return nil, ErrGreylisted
 	}
 
-	err = m.storage.Put(peerOverlayKey(senderOverlay, incomingTx), &overlayVerification{
-		TimeStamp: m.timeNow(),
-		Verified:  false,
-	})
-	if err != nil {
-		return nil, err
-	}
-
 	nTx, isPending, err := m.backend.TransactionByHash(ctx, incomingTx)
 	if err != nil {
+		err2 := m.storage.Put(peerOverlayKey(senderOverlay, incomingTx), &overlayVerification{
```

Comment:
maybe we can move this to a new function and call it once in a `defer` function in the case of an error?

Reply:
then we cannot return the `err2` to anything and we don't have logger available here without changing more stuff, tests, etc.

Reply:
we can still move this out
`err2 := m.storage.Put(peerOverlayKey(senderOverlay, incomingTx), &overlayVerification{
			TimeStamp: m.timeNow(),
			Verified:  false,
		})
		if err2 != nil {
			return nil, err2
		}`